diwako_cbrn.vr/scripts/cbrn/functions/fn_handleDamage.sqf

```diff
@@ -31,10 +31,22 @@ if (_vehicle isNotEqualTo _unit) then {
     _actualThreat = _actualThreat - (_vehicle getVariable ["cbrn_proofing", 0]);
 };
 
+// when the contamination require gasmask only, consume filter in KAT
+if (_threadLevel > 0 && _threadLevel < 2 && cbrn_kat_enabled) then {
+    _unit setVariable ["kat_chemical_enteredPoison", true];
+} else {
+    _unit setVariable ["kat_chemical_enteredPoison", false];
+};
```

Comment:
```suggestion
// when the contamination require gasmask only, consume filter in KAT
if (cbrn_kat_enabled) then {
    if (_threadLevel > 0 && {_threadLevel < 2}) then {
        _unit setVariable ["kat_chemical_enteredPoison", true];
    } else {
        _unit setVariable ["kat_chemical_enteredPoison", false];
    };
};
```

Does not touch the unit if KAT is not enabled, uses lazy eval because the rest of the script also does.